common/src/main/java/com/wynntils/features/ItemScreenshotFeature.java

```diff
@@ -0,0 +1,376 @@
+import com.wynntils.core.features.properties.Stability;
+import com.wynntils.mc.event.InventoryKeyPressEvent;
+import com.wynntils.mc.event.ItemTooltipRenderEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.mc.utils.keybinds.KeyHolder;
+import com.wynntils.mc.utils.keybinds.KeyManager;
+import com.wynntils.wc.utils.WynnUtils;
+import java.awt.*;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.Transferable;
+import java.awt.datatransfer.UnsupportedFlavorException;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(stability = Stability.INVARIABLE, gameplay = GameplayImpact.MEDIUM, performance = PerformanceImpact.SMALL)
+public class ItemScreenshotFeature extends Feature {
+
+    private static Pattern ITEM_PATTERN = Pattern.compile("(Normal|Set|Unique|Rare|Legendary|Fabled|Mythic) Item.*");
+    private Slot screenshotSlot = null;
+
+    private final KeyHolder itemScreenshotKeybind =
+            new KeyHolder("Screenshot Item", GLFW.GLFW_KEY_F4, "Wynntils", true, () -> {});
+
+    @Override
+    protected void onInit(ImmutableList.Builder<Condition> conditions) {}
+
+    @Override
+    protected boolean onEnable() {
+        WynntilsMod.getEventBus().register(this);
+        KeyManager.registerKeybind(itemScreenshotKeybind);
+        return true;
+    }
+
+    @Override
+    protected void onDisable() {
+        WynntilsMod.getEventBus().unregister(this);
+        KeyManager.unregisterKeybind(itemScreenshotKeybind);
+    }
+
+    @Override
+    public MutableComponent getNameComponent() {
+        return new TranslatableComponent("feature.wynntils.itemScreenshot.name");
+    }
+
+    @SubscribeEvent
+    public void onKeyPress(InventoryKeyPressEvent e) {
+        if (itemScreenshotKeybind.getKeybind().matches(e.getKeyCode(), e.getScanCode()))
+            screenshotSlot = e.getHoveredSlot();
+    }
+
+    @SubscribeEvent
+    public void render(ItemTooltipRenderEvent e) {
+        if (screenshotSlot == null) return;
+        // has to be called during a render period
```

Comment:
What does this mean? If you don't call it during a render, it doesn't work? If so, does it crash? `RenderSystem` had some thread assert functions, those might be helpful to check and return if we are not on the correct thread?

Reply:
In my testing, doing all the render stuff directly from the keypress event just didn't affect the framebuffer at all. I can look into more of the RenderSystem stuff.